projects/models.py

```diff
@@ -64,6 +65,40 @@ def annotate_posts_count(self):
             )
         )
 
+    def annotate_leaderboard_questions_count(self):
```

Comment:
Please don't merge for now. Need to check performance

Reply:
It's pretty much the same as the `annotate_posts_count` above on line 46.
And depending on project type only one or the other will be executed.